app/src/main/kotlin/io/homeassistant/companion/android/settings/views/SettingsRow.kt

```diff
@@ -25,13 +25,55 @@ import androidx.compose.ui.unit.dp
 import com.mikepenz.iconics.compose.Image
 import com.mikepenz.iconics.typeface.IIcon
 
+/**
+ * A Composable that displays a typical Material Design clickable list item
+ * with a title, subtitle, and icon slot (from the MDI library).
+ */
 @Composable
 fun SettingsRow(
     primaryText: String,
     secondaryText: String,
     mdiIcon: IIcon?,
     enabled: Boolean,
     onClicked: () -> Unit,
+) {
+    SettingsRow(
+        primaryText = primaryText,
+        secondaryText = secondaryText,
+        icon = {
+            if (mdiIcon != null) {
+                Image(
+                    asset = mdiIcon,
+                    modifier = Modifier
+                        .size(24.dp)
+                        .alpha(if (enabled) ContentAlpha.high else ContentAlpha.disabled),
+                    colorFilter = ColorFilter.tint(
+                        if (enabled) {
+                            MaterialTheme.colors.primary
+                        } else {
+                            contentColorFor(backgroundColor = MaterialTheme.colors.background)
+                        },
+                    ),
+                )
+            } else {
+                Spacer(modifier = Modifier.width(24.dp))
```

Comment:
extract 24.dp into a variable if one day we change the size of the icon it would need to be adjusted.